eva-remapping-get-source/src/main/java/uk/ac/ebi/eva/remapping/source/batch/io/DbsnpSubmittedVariantMongoReader.java

```diff
@@ -31,17 +34,24 @@
 
     public static final String PROJECT_KEY = "study";
 
-    public DbsnpSubmittedVariantMongoReader(String assemblyAccession, MongoTemplate mongoTemplate) {
-        setTemplate(mongoTemplate);
-        setTargetType(DbsnpSubmittedVariantEntity.class);
-        setQuery(new Query(where(REFERENCE_SEQUENCE_FIELD).is(assemblyAccession)));
-    }
+    public static final String TAXONOMY_KEY = "tax";
 
     public DbsnpSubmittedVariantMongoReader(String assemblyAccession, MongoTemplate mongoTemplate,
-                                            List<String> studies) {
+                                          List<String> projects, int taxonomy) {
         setTemplate(mongoTemplate);
         setTargetType(DbsnpSubmittedVariantEntity.class);
-        setQuery(new Query(where(REFERENCE_SEQUENCE_FIELD).is(assemblyAccession).and(PROJECT_KEY).in(studies)));
+
+        Criteria criteria = where(REFERENCE_SEQUENCE_FIELD).is(assemblyAccession);
+
+        if (!CollectionUtils.isEmpty(projects)) {
+            criteria.and(PROJECT_KEY).in(projects);
+        }
+
+        if (taxonomy != 0) {
+            criteria.and(TAXONOMY_KEY).in(taxonomy);
```

Comment:
Does the `in` operator work if this is not a list ?

Reply:
It does work but for clarity I modified the reader to use `is` instead of `in`